### PR TITLE
DEV: Convert second-factor-edit modals to component-based API

### PR DESCRIPTION
**Pull request by janzenisaac**
This PR converts the following modals:
- `second-factor-edit-security-key`
- `second-factor-edit`

to make use of the new component-based API

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/modal/second-factor-edit-security-key.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          return;
+        }
+        this.args.model.markDirty();
+      })
+      .catch((error) => {
+        this.args.model.onError(error);
+      })
+      .finally(() => {
+        this.loading = false;
+        this.modal.close();
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/modal/second-factor-edit-security-key.js:31]:
```suggestion
        this.args.closeModal();
```
(and remove service injection)

File: app/assets/javascripts/discourse/app/components/modal/second-factor-edit.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.args.model.secondFactor.method
+      )
+      .then((response) => {
+        if (response.error) {
+          return;
+        }
+        this.args.model.markDirty();
+      })
+      .catch((error) => {
+        this.modal.close();
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/modal/second-factor-edit.js:28]:
```suggestion
        this.args.closeModal();
```

File: app/assets/javascripts/discourse/app/components/modal/second-factor-edit.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.args.model.onError(error);
+      })
+      .finally(() => {
+        this.loading = false;
+        this.modal.close();
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/modal/second-factor-edit.js:33]:
```suggestion
        this.args.closeModal();
```
(and remove service injection)

**Review by davidtaylorhq — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/modal/second-factor-edit-security-key.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+
+export default class SecondFactorEditSecurityKey extends Component {
+  @tracked loading = false;
+  onClose = this.args.model.onClose;
```
Comment (janzenisaac) [inline app/assets/javascripts/discourse/app/components/modal/second-factor-edit-security-key.js:7]: @davidtaylorhq I am caching the `onClose` argument as we lose context of the args when in the `willDestroy` hook.

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**